pkg/util/winutil/process.go

```diff
@@ -80,9 +84,52 @@ func NtQueryInformationProcess(h windows.Handle, class PROCESSINFOCLASS, target,
 		uintptr(0))
 	if r != 0 {
 		err = windows.GetLastError()
-		return
+		return err
 	}
-	return
+	return nil
+}
+
+// ZwQueryInformationProcess wraps the Windows NT kernel call of the same name.
+func ZwQueryInformationProcess(h windows.Handle, class PROCESSINFOCLASS, target uintptr, size uintptr) (err error) {
+	r, _, _ := procZwQueryInformationProcess.Call(uintptr(h),
+		uintptr(class),
+		target,
+		size,
+		uintptr(0))
+	if r != 0 {
+		err = windows.GetLastError()
+		return err
+	}
+	return nil
+}
+
+// IsProcessProtected checks if the process has any level of protection and returns true if any protection is present
+// more info https://learn.microsoft.com/en-us/windows/win32/procthread/zwqueryinformationprocess
+func IsProcessProtected(h windows.Handle) (bool, error) {
+	// ZwQueryInformationProcess returns 1 byte of protection information when passed in with param 61
+	var processProtection uint8
+	err := ZwQueryInformationProcess(h, ProcessProtectionInformation, uintptr(unsafe.Pointer(&processProtection)), unsafe.Sizeof(processProtection))
+	if err != nil {
+		return false, err
+	}
+	/*
+		The first 3 bits contain the type of protected process:
+
+		typedef enum _PS_PROTECTED_TYPE {
+			PsProtectedTypeNone = 0,
+			PsProtectedTypeProtectedLight = 1,
+			PsProtectedTypeProtected = 2
+		} PS_PROTECTED_TYPE, *PPS_PROTECTED_TYPE;
+
+		if the protection type is anything other than 0, then we cannot open the process with PROCESS_VM_READ
+	*/
+
+	// hex 0x07 == binary 111, so we do a binary AND to determine the protection level
+	if processProtection&0x07 != 0 {
```

Comment:
```suggestion
	if processProtection & 0b111 != 0 {
```
Suggestion: This removes the need for the comment

Reply:
cleaner thanks